drivers/espi/espi_mchp_xec_v2.c

```diff
@@ -1301,6 +1348,103 @@ static void espi_xec_connect_irqs(const struct device *dev)
 #endif
 }
 
+/* MSVW is a 96-bit register and SMVW is a 64-bit register.
+ * Each MSVW/SMVW controls a group of 4 eSPI virtual wires.
+ * Host index located in b[7:0]
+ * Reset source located in b[9:8]
+ * Reset VW values SRC[3:0] located in b[15:12].
+ * MSVW current VW state values located in bits[64, 72, 80, 88]
+ * SMVW current VW state values located in bits[32, 40, 48, 56]
+ */
+static void xec_vw_cfg_properties(const struct xec_signal *p, uint32_t regaddr, uint8_t dir)
+{
+	uint32_t src_ofs = 4u;
+	uint8_t src_pos = (8u * p->bit);
+	uint8_t rst_state = (p->flags >> MCHP_DT_ESPI_VW_FLAG_RST_STATE_POS)
+				& MCHP_DT_ESPI_VW_FLAG_RST_STATE_MSK0;
+	uint8_t rst_src = rst_src = (p->flags >> MCHP_DT_ESPI_VW_FLAG_RST_SRC_POS)
+				& MCHP_DT_ESPI_VW_FLAG_RST_SRC_MSK0;
+
+	if (dir) {
+		src_ofs = 8u;
+	}
+
+	if (rst_state || rst_src) { /* change reset source or state ? */
+		sys_write8(0, regaddr); /* disable register */
+
+		uint8_t temp = sys_read8(regaddr + 1u);
+
+		if (rst_state) { /* change reset state and default value of this vwire? */
+			rst_state--;
+			if (rst_state) {
+				temp |= BIT(p->bit + 4);
+				sys_set_bit(regaddr + src_ofs, src_pos);
+			} else {
+				temp |= ~BIT(p->bit + 4);
+				sys_clear_bit(regaddr + src_ofs, src_pos);
+			}
+		}
+
+		if (rst_src) { /* change reset source of all vwires in this group? */
+			rst_src--;
+			temp = (temp & ~0x3u) | (rst_src & 0x3u);
+		}
+
+		sys_write8(temp, regaddr + 1u);
+	}
+
+	if (sys_read8(regaddr) != p->host_idx) {
+		sys_write8(p->host_idx, regaddr);
+	}
+}
+
+/* Check each VW register set host index is present.
+ * Some VW's power up with the host index and others do not.
+ * NOTE: Virtual wires are in groups of 4. Disabling one wire in a group
+ * will disable all wires in the group. We do not implement disabling.
+ */
+static void xec_vw_config(const struct device *dev)
+{
+	for (int i = 0; i < ARRAY_SIZE(vw_tbl); i++) {
```

Comment:
@MartinYan-mchp @jvasanth1 this should not start zero but only take care of the customizable or platform-specific virtual wires.

The rest of eSPI VW have well defined initial, host index mappings. 
Allowing to change them via device tree is probably in detriment may be not best approach.

Reply:
should we start this config from` ESPI_VWIRE_SIGNAL_SLV_GPIO_0` and keep the remaining (spec-defined) VW hard-coded in the driver (such that they don't come from the DTS)

Reply:
updated to start iterating the for loop from `ESPI_VWIRE_SIGNAL_SLV_GPIO_0` instead of 0